src/main/scala/com/algorithm/LocalMaximaColouring.scala

```diff
@@ -7,71 +7,45 @@ case class Colour(num: Int) {
   }
 }
 
-trait LocalMaximalColouringAbstractMode extends VertexProgram[Int, Int, Set[Int], Colour] {
-  override def gather(edgeVal: Int, message: Int): Set[Int] = Set(message)
-
-  override def sum(a: Set[Int], b: Set[Int]): Set[Int] = a.union(b)
+object Colour {
+  val Blank = Colour(-1)
+}
 
-  override def apply(
-      superStepNumber: Int,
-      thisVertex: VertexInfo,
-      oldVal: Colour,
-      total: Option[Set[Int]]
-  ): Colour = {
-    if (superStepNumber == 0) {
-      Colour(-1)
-    } else {
-      oldVal match {
-        case Colour(-1) => {
-          total match {
-            case None => {
-              // Colour myself with superstep number
-              Colour(superStepNumber - 1)
-            }
-            case Some(idSet) => {
-              if (idSet.max < thisVertex.id) {
-                // Colour myself with superstep number
-                Colour(superStepNumber - 1)
-              } else {
-                Colour(-1)
-              }
-            }
-          }
-        }
-        case c => c
-      }
-    }
+object LocalMaximaColouring extends VertexProgram[Int, Int, Int, Colour] {
+  
+  // Map messages into accumulator values
+  override def gather(edgeVal: Int, message: Int): Int = message
+
+  // Combine accumulator values
+  override def sum(a: Int, b: Int): Int = Math.max(a, b)
+
+  // Compute new local state for this vertex based on accumulated sum
+  override def apply(superStepNumber: Int, thisVertex: VertexInfo, oldVal: Colour, total: Option[Int]): Colour = {
```

Comment:
For consistency, if you want to keep longer line lengths then maybe also change the .scalafmt.conf to match your desired limit